src/WebApi/Controllers/UsersController.cs

```diff
@@ -210,6 +210,37 @@ public Task<ActionResult<Result<CharacterCharacteristicsViewModel>>> GetCharacte
         }));
     }
 
+    /// <summary>
+    /// Get character competitive rating.
+    /// </summary>
+    /// <param name="id">Character id.</param>
+    /// <returns>The character rating.</returns>
+    /// <response code="200">Ok.</response>
+    [HttpGet("self/characters/{id}/competitive-rating")]
+    public Task<ActionResult<Result<CharacterCompetitiveRatingViewModel>>> GetCharacterRating([FromRoute] int id)
+    {
+        return ResultToActionAsync(Mediator.Send(new GetUserCharacterCompetitiveRatingQuery
+        {
+            UserId = CurrentUser.User!.Id,
+            CharacterId = id,
+        }));
+    }
+
+    /// <summary>
+    /// Get top character competitive ratings.
+    /// </summary>
+    /// <returns>The top character competitive ratings.</returns>
+    /// <response code="200">Ok.</response>
+    [HttpGet("leaderboard")]
```

Comment:
The path is generated using the controller path + the value here. So it's `users/leaderboard`. I would suggest adding a new Controller named LeaderboardController instead.